src/files/manager.ts

```diff
@@ -0,0 +1,153 @@
+import createDebug from 'debug';
+
+import { HttpClient } from '../http';
+import { URLHelper } from '../utilities';
+import { validateFileQueryParams, validateFileId } from '../validators';
+
+import { FileQueryParams, FileListResponse, FileResponse } from './types';
+
+const debug = createDebug('strapi:files');
+
+/**
+ * A service class designed for interacting with the file management API in a Strapi application.
+ *
+ * It provides methods to fetch and retrieve files uploaded through the Strapi Media Library.
+ * Note that the Strapi upload plugin API uses a different response format than the content API,
+ * returning file data as a flat array rather than the typical data/meta structure.
+ */
+export class FilesManager {
+  private readonly _httpClient: HttpClient;
+
+  /**
+   * Creates an instance of FilesManager.
+   *
+   * @param httpClient - An instance of HttpClient to handle HTTP communication.
+   *
+   * @example
+   * ```typescript
+   * const httpClient = new HttpClient('http://localhost:1337/api');
+   * const filesManager = new FilesManager(httpClient);
+   * ```
+   */
+  constructor(httpClient: HttpClient) {
+    this._httpClient = httpClient;
+
+    debug('initialized files manager');
+  }
+
+  /**
+   * Retrieves a list of files based on optional query parameters.
+   *
+   * @param queryParams - Optional parameters to filter or sort the results.
+   * @returns A promise that resolves to an array of file objects.
+   *
+   * @throws {HTTPError} if the HTTP client encounters connection issues, the server is unreachable, or authentication fails.
+   * @throws {Error} if invalid query parameters are provided.
+   *
+   * @example
+   * ```typescript
+   * const filesManager = new FilesManager(httpClient);
+   *
+   * const files = await filesManager.find({
+   *   filters: { mime: { $contains: 'image' } },
+   *   sort: 'name:asc',
+   * });
+   *
+   * console.log(files); // Array of file objects
+   * ```
+   */
+  async find(queryParams?: FileQueryParams): Promise<FileListResponse> {
+    debug('finding files');
+
+    try {
+      // Validate query parameters before making the request
+      if (queryParams) {
+        validateFileQueryParams(queryParams);
+      }
+
+      let url = '/upload/files';
```

Comment:
Could we extract that to a constant (`./constants.ts`) ?

Also is this route customizable at all in Strapi or will it always be the same? (just wondering)